genesyscloud/outbound_filespecificationtemplate/resource_genesyscloud_outbound_filespecificationtemplate_test.go

```diff
@@ -0,0 +1,365 @@
+					resource.TestCheckResourceAttr(resourceName+"."+resourceId, "column_information.1.length", column2LengthUpdated),
+				),
+			},
+			{
+				ResourceName:      resourceName + "." + resourceId,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+		CheckDestroy: testVerifyOutboundFileSpecificationTemplateDestroyed,
+	})
+}
+
+func generateOutboundFileSpecificationTemplate(
+	resourceId string,
+	name string,
+	description string,
+	format string,
+	numberOfHeaderLinesSkipped string,
+	numberOfTrailerLinesSkipped string,
+	header string,
+	delimiter string,
+	delimiterValue string,
+	nestedBlocks ...string,
+) string {
+	if description != "" {
+		description = fmt.Sprintf(`description = "%s"`, description)
+	}
+	if format != "" {
+		format = fmt.Sprintf(`format = "%s"`, format)
+	}
+	if numberOfHeaderLinesSkipped != "" {
+		numberOfHeaderLinesSkipped = fmt.Sprintf(`number_of_header_lines_skipped = "%s"`, numberOfHeaderLinesSkipped)
+	}
+	if numberOfTrailerLinesSkipped != "" {
+		numberOfTrailerLinesSkipped = fmt.Sprintf(`number_of_trailer_lines_skipped = "%s"`, numberOfTrailerLinesSkipped)
+	}
+	if header != "" {
+		header = fmt.Sprintf(`header = "%s"`, header)
+	}
+	if delimiter != "" {
+		delimiter = fmt.Sprintf(`delimiter = "%s"`, delimiter)
+	}
+	if delimiterValue != "" {
+		delimiterValue = fmt.Sprintf(`delimiter_value = "%s"`, delimiterValue)
+	}
+	return fmt.Sprintf(`
+	resource "genesyscloud_outbound_filespecificationtemplate" "%s" {
+		name = "%s"
+		%s
+		%s
+		%s
+		%s
+		%s
+		%s
+		%s
+		%s
+	}
+	`,
+		resourceId,
+		name,
+		description,
+		format,
+		numberOfHeaderLinesSkipped,
+		numberOfTrailerLinesSkipped,
+		header,
+		delimiter,
+		delimiterValue,
+		strings.Join(nestedBlocks, "\n"))
+}
+
+func generateOutboundFileSpecificationTemplateColumnInformation(
+	columnName string,
+	columnNumber string,
+	startPosition string,
+	length string,
+) string {
+	if columnName != "" {
+		columnName = fmt.Sprintf(`column_name = "%s"`, columnName)
+	}
+	if columnNumber != "" {
+		columnNumber = fmt.Sprintf(`column_number = "%s"`, columnNumber)
+	}
+	if startPosition != "" {
+		startPosition = fmt.Sprintf(`start_position = "%s"`, startPosition)
+	}
+	if length != "" {
+		length = fmt.Sprintf(`length = "%s"`, length)
+	}
+	return fmt.Sprintf(`
+	column_information {
+		%s
+		%s
+		%s
+		%s
+	}
+`, columnName, columnNumber, startPosition, length)
+}
+
+func generateOutboundFileSpecificationTemplatePreprocessingRule(
+	find string,
+	replaceWith string,
+	global string,
+	ignoreCase string,
+) string {
+	if find != "" {
+		find = fmt.Sprintf(`find = "%s"`, find)
+	}
+	if replaceWith != "" {
+		replaceWith = fmt.Sprintf(`replace_with = "%s"`, replaceWith)
+	}
+	if global != "" {
+		global = fmt.Sprintf(`global = "%s"`, global)
+	}
+	if ignoreCase != "" {
+		ignoreCase = fmt.Sprintf(`ignore_case = "%s"`, ignoreCase)
+	}
+	return fmt.Sprintf(`
+	preprocessing_rule {
+		%s
+		%s
+		%s
+		%s
```

Comment:
I would recommend this approach:

```golang
return fmt.Sprintf(`
preprocessing_rule {
    ...
    ignore_case = %s
}
`, ignoreCase)
```
If ignore_case has a value - pass `strconv.Quote(varThatHoldsValueOfIgnoreCase)` to `generateOutboundFileSpecificationTemplatePreprocessingRule` , so that is appears as a quoted value in the returned string
If it does not - pass in `genesyscloud.NullValue` 

There is other code in the codebase (that I wrote long ago) done the way you have it above, but I don't think it's very readable, so I'd rather we move away from it.